### PR TITLE
setup devise

### PR DESCRIPTION
**Pull request by AntonellaMorittu**
#### Description
Changes proposed in this pull request:

* Devise setup

#### What I have learned working on this feature: 
How  to setup devise in my app

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (diraulo) [inline spec/models/user_spec.rb:3]: Please add some unit tests for the `User` model

**Review by diraulo — CHANGES_REQUESTED** [review]
I see no tests in this PR.

You've created some new step definitions for the sign_up feature but no feature files were created with scenarios? (forgot to commit that file perhaps?)

The `User` model has no unit tests either...

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+end
```
Comment (magnus-thor) [inline spec/models/user_spec.rb:4]: There should never be a empty spec file, In the least test the creation of a user like this
```
  it 'should have valid Factory' do
    expect(FactoryBot.create(:user)).to be_valid
  end
```

File: spec/models/user_spec.rb (hunk 1 above)
Comment (NellaMori) [reply spec/models/user_spec.rb:3]: They are not on this branch @diraulo, we are waiting for this PR to be merged in order to proceed implementing the signup function. This is just a setup branch..

Comment (diraulo) [reply spec/models/user_spec.rb:3]: The thing is the moment you add the devise gem to your application and do the following as per devise's documentation:

- setup a user model
- add devise routes `devise_for: users`
- etc.

That's practically all it takes to have every authentication actions setup already (Sign up, Sign in, etc...)

If you check your routes you'll see that you have routes for all of them, if you fire up your application server, you can sign-up and sign-in (all the forms are already created for you by devise)

The user model should be fairly straightforward to test, if you check how we tested previous models, and look at the schema file, you'll be able to see what fields you should check for in your unit tests.

**Comment by tochman** [comment]
You created a User model in that PR. That model needs to be tested. Start with that please

File: spec/models/user_spec.rb (hunk 2 above)
Comment (diraulo) [reply spec/models/user_spec.rb:4]: And we can also test for table columns in this spec

**Comment by diraulo** [comment]
@AntonellaMorittu Any specific reasons as to why this PR was closed? 🤔

**Comment by NellaMori** [comment]
@diraulo It has been closed because the sign-up and log-in buttons were not set and we couldn't understand which step we did wrong. Now it should be okay 😃  https://github.com/CraftAcademy/broke_no_more/pull/10

**Pull request closed without merging**